src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java

```diff
@@ -288,9 +288,81 @@ boolean canAddField() {
       }
     }
   }
+
+  /**
+   * Filters db documents based on non document id field values if id use "$in" operator, use
+   * IDFilter, since partition key 'or' is not supported besides id, "$in" operator, use INFilter
+   */
+  public static class InFilter extends DBFilterBase {
+    private final Object arrayValue;
+    protected final InFilter.Operator operator;
+
+    @Override
+    JsonNode asJson(JsonNodeFactory nodeFactory) {
+      return DBFilterBase.getJsonNode(nodeFactory, arrayValue);
+    }
+
+    @Override
+    boolean canAddField() {
+      return false;
+    }
+
+    public enum Operator {
```

Comment:
Will we have other Operator values at some point? Seems strange to define an Enum with just 1 value.

Reply:
sure, I am following other filter's definition. Let me add some doc.

Reply:
But those other filters probably have multiple values (or could conceivably get more)?
If there won't be others, I don't think separate type should be added.

Reply:
This class is weird, too (not fault of this PR :) ) -- `DBFilterBase` is a base class, yet there are nested sub-classes that implement it for some reason. That seems a bit wrong.
So maybe eventually these would be good to refactor out of containing parent class.
(not in this PR of course)